calculator_test.go

```diff
@@ -3,27 +3,125 @@ package calculator_test
 import (
 	"calculator"
 	"math"
+	"math/rand"
 	"testing"
+	"time"
 )
 
+func almostEqual(a, b float64) bool {
+	const float64EqualityThreshold = 1e-9
+	return math.Abs(a-b) <= float64EqualityThreshold
+}
+
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	var tcs = []struct {
+		name string
+		want float64
+		a    float64
+		b    float64
+	}{
+		{
+			name: "Two positive numbers",
+			want: 5,
+			a:    3,
+			b:    2,
+		},
+		{
+			name: "Zero subtracted with any number",
+			want: -2,
+			a:    0,
+			b:    -2,
+		},
+		{
+			name: "Positive and negative number",
+			want: -1,
+			a:    -5,
+			b:    4,
+		},
+		{
+			name: "Two negative numbers",
+			want: -11,
+			a:    -8,
+			b:    -3,
+		},
+		{
+			name: "Two fractional numbers",
+			want: 6.5,
+			a:    3.3,
+			b:    3.2,
+		},
+	}
+
+	for _, tc := range tcs {
+		got := calculator.Add(tc.a, tc.b)
+		if !almostEqual(tc.want, got) {
+			t.Errorf("Test Add(%f, %f) (%s) failed: want %f, got %f",
```

Comment:
```suggestion
			t.Errorf("%s: Add(%f, %f): want %f, got %f",
```

We don't need to repeat `Test` here, or `failed`, since we wouldn't see this output at all except in the context of a failing test!